pkg/source/image.go

```diff
@@ -139,28 +146,54 @@ func (i *Image) getDesiredPodApplyConfig(bundle *rukpakv1alpha2.BundleDeployment
 				WithImage(i.UnpackImage).
 				WithImagePullPolicy(corev1.PullIfNotPresent).
 				WithCommand("/cp", "-Rv", "/unpack", "/util/bin/unpack").
-				WithVolumeMounts(applyconfigurationcorev1.VolumeMount().
-					WithName("util").
-					WithMountPath("/util/bin"),
-				).
+				WithVolumeMounts(func() []*applyconfigurationcorev1.VolumeMountApplyConfiguration {
+					var volumeMounts []*applyconfigurationcorev1.VolumeMountApplyConfiguration
+					if gocoverdirEnv != "" {
+						volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+							WithName("test-coverage").
+							WithMountPath(gocoverdirEnv))
+					}
+					volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+						WithName("util").
+						WithMountPath("/util/bin"))
+					return volumeMounts
+				}()...).
+				WithEnv(applyconfigurationcorev1.EnvVar().WithName("GOCOVERDIR").WithValue(gocoverdirEnv)).
 				WithSecurityContext(containerSecurityContext).
 				WithTerminationMessagePolicy(corev1.TerminationMessageFallbackToLogsOnError),
 			).
 			WithContainers(applyconfigurationcorev1.Container().
 				WithName(imageBundleUnpackContainerName).
 				WithImage(bundle.Spec.Source.Image.Ref).
 				WithCommand("/bin/unpack", "--bundle-dir", "/").
-				WithVolumeMounts(applyconfigurationcorev1.VolumeMount().
-					WithName("util").
-					WithMountPath("/bin"),
-				).
+				WithVolumeMounts(func() []*applyconfigurationcorev1.VolumeMountApplyConfiguration {
+					var volumeMounts []*applyconfigurationcorev1.VolumeMountApplyConfiguration
+					if gocoverdirEnv != "" {
+						volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+							WithName("test-coverage").
+							WithMountPath(gocoverdirEnv))
+					}
+					volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+						WithName("util").
+						WithMountPath("/bin"))
+					return volumeMounts
+				}()...).
+				WithEnv(applyconfigurationcorev1.EnvVar().WithName("GOCOVERDIR").WithValue(gocoverdirEnv)).
```

Comment:
Can we only set this if `gocoverdirEnv != ""`?

Reply:
Passing an empty configuration is causing errors. Tried various options, but looks like the only one is to create the entire container spec separately based on whether env var is present or not. 

Leaving it to be as is for now, if the env is not set it results in an empty string which doesn't cause a problem for now. Also this is just a band aid, and will replaced by direct registry client anyways.